app/controlplane/internal/service/attestation.go

```diff
@@ -213,9 +213,17 @@ func (s *AttestationService) Store(ctx context.Context, req *cpAPI.AttestationSe
 		return nil, errors.NotFound("not found", "robot account not found")
 	}
 
-	// Check that provided workflowRun belongs to workflow encoded in the robot account
-	if exists, err := s.wrUseCase.ExistsInWorkflow(ctx, robotAccount.WorkflowID, req.WorkflowRunId); err != nil || !exists {
-		return nil, errors.NotFound("not found", "workflowRun not found")
+	if robotAccount.WorkflowID == "" {
+		wr, err := s.wrUseCase.GetByIDInOrgOrPublic(ctx, robotAccount.OrgID, req.WorkflowRunId)
```

Comment:
I do not think this method is what we want. As the name suggests, this will return a run  that belongs to you or it's public, we do not want you to be able to attest public workflows.

The point of the check is to make sure that the run is from one of your orgs.